org.ekstep.questionset-1.0/renderer/plugin.js

```diff
@@ -46,88 +46,95 @@ org.ekstep.questionsetRenderer = IteratorPlugin.extend({ // eslint-disable-line
+      QSTelemetryLogger.qsConfig = this._questionSetConfig;
+      if(data.isQuestionPreview){
+        // get navigation plugin instance & empty all customNavigation object of it
+        org.ekstep.pluginframework.pluginManager.plugins['org.ekstep.navigation'].p.prototype._customNavigationPlugins=[]
+      }
+      // this.setupNavigation();
+      // Get all questions in the question set
+      var quesArray = JSON.parse(JSON.stringify(data[this._constants.questionPluginId]));
+      //if question set have one question then convert from object to array for device issue
+      this._masterQuestionSet = _.isArray(quesArray) ? quesArray : [quesArray];
+      // If this isn't the first time the question set is being rendered, restore its earlier state
+      this._questionStates = {};
+      this._renderedQuestions = [];
+      var question = undefined;
+      var savedQSState = this.getQuestionSetState();
+  
+      EkstepRendererAPI.addEventListener("renderer:plugin:reset", function(e) {
+        this.reInstateQuestionsOnReview(e.target.data);
+      }, this);
+      
+      var savedCurrentQuestion = this.questionExistInQS(savedQSState);
+      if (savedQSState && savedCurrentQuestion) {
+        this._renderedQuestions = savedQSState.renderedQuestions;
+        question = savedQSState.currentQuestion;
+        this._questionStates = savedQSState.questionStates;
+        this._currentQuestionState = this.getQuestionState(question.id);
+        this._itemIndex = savedQSState.itemIndex >= 0 ? savedQSState.itemIndex : -1;
+      } else {
+        question = this.getNextQuestion();
+      }
+      if(this._itemIndex > 0){
+          EventBus.dispatch("renderer:previous:enable");
+      }
+  
+      // Register for navigation hooks
+      this.registerNavigation(instance);
+  
+      this.saveQuestionSetState();
+      // Render the question
+      this.renderQuestion(question);
 
-    EkstepRendererAPI.addEventListener("renderer:plugin:reset", function(e) {
-      this.reInstateQuestionsOnReview(e.target.data);
-    }, this);
-    
-    var savedCurrentQuestion = this.questionExistInQS(savedQSState);
-    if (savedQSState && savedCurrentQuestion) {
-      this._renderedQuestions = savedQSState.renderedQuestions;
-      question = savedQSState.currentQuestion;
-      this._questionStates = savedQSState.questionStates;
-      this._currentQuestionState = this.getQuestionState(question.id);
-      this._itemIndex = savedQSState.itemIndex >= 0 ? savedQSState.itemIndex : -1;
-    } else {
-      question = this.getNextQuestion();
-    }
-    if(this._itemIndex > 0){
-        EventBus.dispatch("renderer:previous:enable");
+  } 
+  catch(e) {
+      org.ekstep.pluginframework.eventManager.dispatchEvent('plugin:error', { plugin: manifest.id, version: manifest.ver, action: 'load', err: e })
+      TelemetryService.error({ 'err': e.error, 'errtype': 'SYSTEM', 'stacktrace': e.stackTrace, 'plugin': e.pluginType }); 
```

Comment:
1. Please fix indentation issues
2. Error telemetry plugin should object & e.pluginType is not valid please check and fix